dashboard/src/shared/Namespace.ts

```diff
@@ -1,31 +1,38 @@
 import { axiosWithAuth } from "./AxiosInstance";
-import { APIBase } from "./Kube";
 import * as url from "./url";
 
 import { ForbiddenError, IResource, NotFoundError } from "./types";
 
 export default class Namespace {
-  public static async list() {
+  public static async list(cluster: string) {
+    // This call is hitting an actual backend endpoint (see pkg/http-handler.go)
+    // while the other calls (create, get) are hitting the k8s API via the
+    // frountend nginx.
     const { data } = await axiosWithAuth.get<{ namespaces: IResource[] }>(
-      url.backend.namespaces.list(),
+      url.backend.namespaces.list(cluster),
     );
     return data;
   }
 
-  public static async create(name: string) {
-    const { data } = await axiosWithAuth.post<IResource>(`${APIBase}/api/v1/namespaces/`, {
-      apiVersion: "v1",
-      kind: "Namespace",
-      metadata: {
-        name,
+  public static async create(cluster: string, name: string) {
+    const { data } = await axiosWithAuth.post<IResource>(
+      `api/clusters/${cluster}/api/v1/namespaces/`,
```

Comment:
this URL should be in the `url` lib

Reply:
I've done this as part of a larger change to improve the direct calls to the k8s api from the dashboard, so they are easily identifiable, as part of #1877 .